### PR TITLE
Address upstream comments to use uuid_t for swidget->uuid

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

**Review by plbossart — DISMISSED** [review]

**Review by plbossart — APPROVED** [review]

**Review by bardliao — DISMISSED** [review]

**Comment by ranj063** [comment]
please done merge this. I need to switch to using guid_t type instead of uuid_t

**Comment by ranj063** [comment]
> please done merge this. I need to switch to using guid_t type instead of uuid_t

fixed now.

**Review by plbossart — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

File: sound/soc/sof/ipc3-topology.c [hunk 1]
```diff
@@ -310,13 +310,12 @@ static const struct sof_token_info ipc3_token_list[SOF_TOKEN_COUNT] = {
...
 			    int index)
 {
-	u8 nil_uuid[SOF_UUID_SIZE] = {0};
 	struct sof_ipc_comp *comp;
 	size_t total_size = *ipc_size;
 	size_t ext_size = sizeof(swidget->uuid);
 
 	/* only non-zero UUID is valid */
-	if (memcmp(swidget->uuid, nil_uuid, SOF_UUID_SIZE))
+	if (guid_is_null(&swidget->uuid))
```
Comment (aiChaoSONG) [inline sound/soc/sof/ipc3-topology.c:318]: You said `uuid_is_null()` is used in commit message, but in code, `guid_is_null()` is used.

**Review by dbaluta — APPROVED** [review]

File: sound/soc/sof/ipc3-topology.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/ipc3-topology.c:318]: again don't pay too much attention for fixup! commit messages @aiChaoSONG, they will disappear when we squash them.

**Review by plbossart — APPROVED** [review]

**Pull request merged**